### PR TITLE
Drop parfive less than 2

### PR DESCRIPTION
**Pull request by nabobalis**
Also a hack for adding the words CI-test to the user agent headers for our tests.

Fixes https://github.com/sunpy/sunpy/issues/6308
Fixes https://github.com/sunpy/sunpy/issues/6940

I adjusted the previous changelog for updated versions instead of a new one.

### CODE REVIEW

File: sunpy/conftest.py [hunk 1]
```diff
@@ -114,6 +114,16 @@ def undo_config_dir_patch():
...
 
 
+@pytest.fixture(scope='session', autouse=True)
+def parfive_header_test(request):
+    """
+    Add a keyword to tell parfive this is a test run.
+    """
+    os.environ["PARFIVE_SUNPY_TESTS"] = "True"
+    yield
+    del os.environ["PARFIVE_SUNPY_TESTS"]
```
Comment (nabobalis) [inline sunpy/conftest.py:124]: Was not sure how to do this, so I did this as it was quick.

Comment (dstansby) [reply sunpy/conftest.py:124]: Seems legit to me 👍

**Review by dstansby — CHANGES_REQUESTED** [review]
- The `oldestdeps` version pin needs updating in `tox.ini`
- There should be a note in the changelog and in the what's new that says something like "We are aware the change in the parfive minimum version is a release earlier than our dependency policy allows for, but due to significant issues that parfive v2 solves we have decided to bump it now"
- Can you explain a bit more about why you've tweaked the user agent for the tests?

File: sunpy/util/parfive_helpers.py [hunk 2]
```diff
@@ -14,36 +14,16 @@
 parfive_version = Version(parfive.__version__)
 sunpy_headers = {
     "User-Agent": f"sunpy/{sunpy.__version__} parfive/{parfive.__version__} "
-    f"aiohttp/{aiohttp.__version__} python/{sys.version[:5]}"
+    f"aiohttp/{aiohttp.__version__} python/{sys.version[:5]} "
 }
+if os.environ.get("PARFIVE_SUNPY_TESTS"):
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/util/parfive_helpers.py:19]: I thought the `get` defaults to a None if the var is missing?

File: changelog/6742.feature.rst [hunk 3]
```diff
@@ -3,6 +3,7 @@ The minimum required versions of several core dependencies have been updated:
 - Python 3.9
 - astropy 5.0.1
 - numpy 1.21.0
+- parfive 2.0.0
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply changelog/6742.feature.rst]: Fixed.

File: sunpy/util/parfive_helpers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dstansby) [reply sunpy/util/parfive_helpers.py:19]: Huh, so it does, my mistake!

**Comment by nabobalis** [comment]
> * Can you explain a bit more about why you've tweaked the user agent for the tests?

The motivation is to help the VSO or other providers to be able to see if requests coming from sunpy come either from our CI or actual users.

This is used to be more of a problem back in the day when we ran a larger online test suite but I think it would still be nice for data providers to be able to parse logs and know if our test suite is causing a problem or if its users.

**Review by dstansby — APPROVED** [review]

File: sunpy/conftest.py (hunk 1 above)
Comment (Cadair) [reply sunpy/conftest.py:124]: Yeah this seems good, but why don't we make this a generic environment var for being in our test suite? We might have other non-parfive uses for it in the future?

Comment (nabobalis) [reply sunpy/conftest.py]: Was not sure what to call it, so I made one up.

**Review by Cadair — APPROVED** [review]

**Pull request merged**